backend/src/payments/payments.service.ts

```diff
@@ -19,69 +28,129 @@ export class PaymentsService {
+        ? ILike(`%${filterArguments.lastName}%`)
+        : undefined;
+
+    return {
+      date: dateProperty,
+      note: noteProperty,
+      trainee: {
+        user: {
+          firstName: firstNameProperty,
+          lastName: lastNameProperty,
+          organizationId,
+          id: userId,
         },
+        id: traineeId,
       },
-      relations: {
-        trainee: { user: true },
-      },
-      order: {
-        date: 'DESC',
-      },
+    };
+  }
+
+  async findAll(presentationArguments?: PaymentPresentationArguments) {
+    const { id: organizationId } =
+      this.organizationDomainService.getRequestOrganization();
+
+    const limitArguments = getLimitArguments(presentationArguments?.pagination);
+
+    const [payments, count] = await this.paymentsRepository.findAndCount({
+      ...limitArguments,
+      order: this.buildOrderOption(presentationArguments?.sort),
+      where: this.buildWhereOption(
+        presentationArguments?.filter,
+        organizationId,
+      ),
     });
 
-    return payments;
+    return { payments, count };
   }
 
-  async findAllByTraineeId(traineeId: number) {
+  async findAllByTraineeId(
+    traineeId: number,
+    presentationArguments?: PaymentPresentationArguments,
+  ) {
     const { id: organizationId } =
       this.organizationDomainService.getRequestOrganization();
 
-    const payments = await this.paymentsRepository.find({
-      where: {
-        trainee: {
-          id: traineeId,
-          user: {
-            organizationId,
-          },
-        },
-      },
-      order: {
-        date: 'DESC',
-      },
+    const limitArguments = getLimitArguments(presentationArguments?.pagination);
+
+    const [payments, count] = await this.paymentsRepository.findAndCount({
+      ...limitArguments,
+      order: this.buildOrderOption(presentationArguments?.sort),
+      where: this.buildWhereOption(
+        presentationArguments?.filter,
+        organizationId,
+        traineeId,
+      ),
     });
 
-    return payments;
+    return { payments, count };
   }
 
-  async findAllByUserId(traineeId: number) {
+  async findAllByUserId(
+    userId: number,
+    presentationArguments?: PaymentPresentationArguments,
+  ) {
     const { id: organizationId } =
       this.organizationDomainService.getRequestOrganization();
 
-    const payments = await this.paymentsRepository.find({
-      where: {
-        trainee: {
-          user: {
-            organizationId,
-            id: traineeId,
-          },
-        },
-      },
-      order: {
-        date: 'DESC',
-      },
+    const limitArguments = getLimitArguments(presentationArguments?.pagination);
+
+    const [payments, count] = await this.paymentsRepository.findAndCount({
+      ...limitArguments,
+      order: this.buildOrderOption(presentationArguments?.sort),
+      where: this.buildWhereOption(
+        presentationArguments?.filter,
+        organizationId,
+        userId,
+      ),
```

Comment:
I think this should be
```suggestion
      where: this.buildWhereOption(
        presentationArguments?.filter,
        organizationId,
        undefined,
        userId,
      ),
```

as the 3rd argument is the `traineeId`